packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -138,13 +205,38 @@ export class Fleet extends Resource implements IFleet {
       throw new Error('baseCapacity must be greater than or equal to 1');
     }
 
-    super(scope, id, { physicalName: props.fleetName });
+    if (
+      props.computeType === FleetComputeType.ATTRIBUTE_BASED &&
+      (!props.computeConfiguration || Object.keys(props.computeConfiguration).length === 0)
+    ) {
+      throw new Error('At least one compute configuration criteria must be specified if computeType is "ATTRIBUTE_BASED"');
+    }
+    if (props.computeConfiguration && props.computeType !== FleetComputeType.ATTRIBUTE_BASED) {
+      throw new Error(`'computeConfiguration' can only be specified if 'computeType' is 'ATTRIBUTE_BASED', got: ${props.computeType}`);
+    }
+
+    const computeConfig = props.computeConfiguration;
+    const diskGiB = computeConfig?.disk?.toGibibytes() ?? 0;
+    const memoryGiB = computeConfig?.memory?.toGibibytes() ?? 0;
+    const vCpu = computeConfig?.vCpu ?? 0;
+
+    this.validatePositiveInteger(diskGiB, 'disk size');
+    this.validatePositiveInteger(memoryGiB, 'memory size');
+    if (!Token.isUnresolved(vCpu)) {
+      this.validatePositiveInteger(vCpu, 'vCPU count');
+    }
```

Comment:
You could move the `Token.isUnresolved` check to `validatePositiveInteger`, it's a little cleaner and it can't hurt the other two to validate them as well

Reply:
You are absolutely right. I have made the corrections!